x-pack/plugins/security_solution/common/detection_engine/schemas/request/rule_schemas.test.ts

```diff
@@ -656,24 +421,20 @@ describe('create rules schema', () => {
   });
 
   test('saved_query type can have filters with it', () => {
-    const payload: CreateRulesSchema = {
-      ...getCreateRulesSchemaMock(),
+    const payload: SavedQueryCreateSchema = {
+      ...getCreateSavedQueryRulesSchemaMock(),
       filters: [],
     };
 
     const decoded = createRulesSchema.decode(payload);
     const checked = exactCheck(payload, decoded);
     const message = pipe(checked, foldLeftRight);
     expect(getPaths(left(message.errors))).toEqual([]);
-    const expected: CreateRulesSchemaDecoded = {
-      ...getCreateRulesSchemaDecodedMock(),
-      filters: [],
-    };
-    expect(message.schema).toEqual(expected);
+    expect(message.schema).toEqual(payload);
   });
 
   test('filters cannot be a string', () => {
-    const payload: Omit<CreateRulesSchema, 'filters'> & { filters: string } = {
+    const payload = {
       ...getCreateRulesSchemaMock(),
       filters: 'some string',
     };
```

Comment:
I think this is ok to do to weaken the types for a test if the Omit feels unergonomic and unwieldily to use to keep the type strict. You should get the majority of type safety from the function `getCreateRulesSchemaMock` still.

If you hover between the two types though of this payload and the `getCreateRulesSchemaMock()` you should notice the interesting thing which is TypeScript during a spread is changing the types slightly.

In regular (outside of tests) we have had some odd behaviors from spreads where TypeScript can change the types slightly or have an ambiguity situation where once we added the type to where we initiated the spread the errors/issues went away.

You might know all of this, just footnotes in case.

Reply:
++ I added back most of the stronger types where the payload is intended to validate correctly or some fields are simply missing. I figure for cases where validation should fail because a field is the wrong type there's not as much benefit so I left them off for convenience.